clientconn.go

```diff
@@ -245,6 +238,12 @@ func DialContext(ctx context.Context, target string, opts ...DialOption) (conn *
 
 	// Determine the resolver to use.
 	cc.parsedTarget = grpcutil.ParseTarget(cc.target)
+	if cc.parsedTarget.Scheme == "" && strings.HasPrefix(cc.target, "unix:") && cc.dopts.copts.Dialer == nil {
```

Comment:
I believe we decided this should be moved into `ParseTarget` with a flag indicating whether a custom dialer had been specified by the user.

Reply:
Made this change.